examples/chart/teleport-kube-agent/templates/delete_hook.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-delete-hook
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-weight": "-4"
+    "helm.sh/hook-delete-policy": hook-succeeded
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Release.Name }}-delete-hook
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-weight": "-3"
+    "helm.sh/hook-delete-policy": hook-succeeded
+rules:
+  - apiGroups: [""]
+    resources: ["secrets",]
+    verbs: ["get", "delete", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ .Release.Name }}-delete-hook
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-weight": "-2"
+    "helm.sh/hook-delete-policy": hook-succeeded
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ .Release.Name }}-delete-hook
+subjects:
+- kind: ServiceAccount
+  name: {{ .Release.Name }}-delete-hook
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-delete-hook
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-delete
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: {{ .Release.Name }}-delete-hook
+    spec:
+      serviceAccountName: {{ .Release.Name }}-delete-hook
+      restartPolicy: OnFailure
+      containers:
+      - name: post-delete-job
+        image: alpine/k8s:1.26.0
```

Comment:
IMO we should change this to use the same image that's specified for `teleport-kube-agent` itself. It can run the exact same `/bin/sh` command using the Teleport image without a problem, and there is a better guarantee that this image will already be accessible/cached within the cluster. For air-gapped deployments (or when people are using private repos like Artifactory etc) I think this will be more likely to work correctly.

Reply:
IIRC we do not ship kubectl in our images.

Reply:
Fair point, although it looks like we were installing it manually before? I guess that wouldn't work on an air-gapped cluster either so it's probably no different.

Reply:
I don't remember seeing kubectl on our public images.

Reply:
`teleport kube-state delete`?

Reply:
that will work when we deploy hardened images.
Are you ok with it @webvictim ?

Reply:
What about the other hook tho? Adding a subcommand to Teleport to delete its state is quite reasonable, adding a subcommand to Teleport to wait for a statefulset and delete a deployment is not, IMO.

Should we just ship kubectl in our images?

Reply:
The other hook will be removed after Teleport 12 because we no longer need to support Teleport 10 upgrades.

Reply:
I implemented `teleport wait` a couple of weeks ago, so the "wait" part is totally doable, we can wait until a service stops resolving IPs (`wait no-resolve`) or add (`wait resolve`).

Reply:
Having a `teleport kube-state` command is also very relevant to my interests because I would need a subcommand to write the CA pins to a secret/configmap so the cluster nodes can join with CA pinning on.

Reply:
Updated with the addition of `teleport kube-state delete` command.